src/libraries/System.Private.Uri/src/System/IriHelper.cs

```diff
@@ -68,114 +94,57 @@ internal static bool CheckIsReserved(char ch, UriComponents component)
         internal static unsafe string EscapeUnescapeIri(char* pInput, int start, int end, UriComponents component)
         {
             int size = end - start;
-            ValueStringBuilder dest = new ValueStringBuilder(size);
-            byte[]? bytes = null;
-
-            int next = start;
-            char ch;
+            ValueStringBuilder dest = size <= 256
+                ? new ValueStringBuilder(stackalloc char[256])
```

Comment:
Shouldn't this stackallock up to `size`?

Reply:
> Shouldn't this stackalloc up to size?

We disable localinit on all dotnet/runtime assemblies, which means it's no more expensive to stackalloc more than less, and it's cheaper for the JIT to stackalloc a const than it is a variable, so our general pattern is what's used here, with some relatively small fixed max.

> Has 256 picked for some reason, or just because it's a nice round number

We've generally used values between 32 and 256... we're unfortunately not particularly consistent on which.